docs/development/justfile.md

```diff
@@ -0,0 +1,769 @@
+just deps-add-dev pytest-benchmark
+```
+
+### Django Commands
+
+```bash
+# Make migrations
+just m makemigrations
+
+# Create superuser
+just m createsuperuser
+
+# Custom management command
+just m my_custom_command
+```
+
+### Celery Development
+
+The `run-dev-celery` command runs all Celery components together with colored output:
+
+- **WORKER** (blue): Main worker for celery and automation_workflow queues
+- **EXPORT** (orange): Export worker
+- **BEAT** (purple): Scheduler with redbeat
+
+Press Ctrl+C to stop all workers.
+
+To adjust log level or pool type:
+
+```bash
+CELERY_LOG_LEVEL=DEBUG just run-dev-celery
+CELERY_POOL=threads just run-dev-celery
+```
+
+## Differences from Docker Development
+
+| Aspect | Justfile (local) | Docker |
+|--------|------------------|--------|
+| Python env | Local venv via uv | Container venv |
+| Services | External (Redis, PostgreSQL) | docker-compose managed |
+| Hot reload | Native file watching | Volume mounts |
+| Debugging | Direct IDE integration | Remote debugging |
+
+For services like Redis and PostgreSQL, either:
+- Run them locally
+- Use `just dc-dev up db redis` to start just the services
+- Use the test database: `just test-db up`
+
+## Shell Completion
+
+### The Tradeoff
+
+`just` provides convenient, documented commands but **does not support shell tab-completion** for command arguments. This is because tab completion happens in your shell *before* the command runs, so the shell doesn't know that `just b test` will eventually run `pytest`.
+
+For example, when you type:
+```bash
+just b test ../enterpr<TAB>   # No completion - shell doesn't know about pytest
+```
+
+If you need tab completion (e.g., for file paths), use the underlying command directly from the `backend/` directory:
+```bash
+source ../.env.local && uv run pytest ../enterpr<TAB>   # Completes to ../enterprise/
+```
+
+### Running Commands Directly
+
+For local development with bash completion, you can bypass `just` and run commands directly. You'll need to:
+
+1. **Load environment variables** from `.env.local` (in project root)
+2. **Use `uv run`** to execute in the venv
+
+From the `backend/` directory:
+
+```bash
+# Option 1: Source env and run (one-liner)
+source ../.env.local && uv run pytest ../enterprise/backend/tests/
+
+# Option 2: Activate venv first
+source ../.venv/bin/activate
+source ../.env.local
+pytest ../enterprise/backend/tests/
+```
+
+### Loading and Clearing Environment Variables
+
+Use these just recipes to manage environment variables in your shell:
+
+```bash
+# Load .env.local into current shell
+eval "$(just env-load)"
+
+# Clear all variables from .env.local
+eval "$(just env-clear)"
+
+# Alternative: start a fresh shell (loses all changes)
+exec bash
+```
+
+**Note:** The recipes output commands that you `eval` because just runs in a subshell and can't modify your parent shell directly.
+
+### When to Use Each Approach
+
+| Use `just` when... | Use direct commands when... |
```

Comment:
Some of the tables like this one looks like AI slop. This doc should describe working with just, not dive into alternatives IMHO. It is already long.